sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -39,8 +39,36 @@ import org.apache.spark.util.{MutableURLClassLoader, Utils}
 
 /**
  * A class that holds all state shared across sessions in a given [[SQLContext]].
+ *
+ * @param sparkContext The Spark context associated with this SharedState
+ * @param initialConfigs The configs from the very first created SparkSession
  */
-private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
+private[sql] class SharedState(
+    val sparkContext: SparkContext,
+    initialConfigs: scala.collection.Map[String, String])
+  extends Logging {
+
+  // This variable should be lazy, because in the first place we need to load hive-site.xml into
+  // hadoopConf and determine the warehouse path which will be set into both spark conf and hadoop
+  // conf avoiding be affected by any SparkSession level options
+  private lazy val (conf, hadoopConf) = {
```

Comment:
shall we just move this code block after `val warehousePath: String ...`? Then we don't need the lazy val.